commands/scan/scan.go

```diff
@@ -360,12 +393,32 @@ func (scanCmd *ScanCommand) createIndexerHandlerFunc(file *spec.File, indexedFil
 					return err
 				}
 				scanResults, err := scangraph.RunScanGraphAndGetResults(scanGraphParams, xrayManager)
+
 				if err != nil {
 					log.Error(fmt.Sprintf("scanning '%s' failed with error: %s", graph.Id, err.Error()))
 					indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
 					return
 				}
-				resultsArr[threadId] = append(resultsArr[threadId], &ScanInfo{Target: filePath, Result: scanResults})
+
+				// flatResults := []xrutils.ScaScanResult{}
+				extendedScanResults := utils.ExtendedScanResults{}
+				if entitledForJas && scanCmd.commandSupportsJAS {
+					depsList := depsListFromVulnerabilities(*scanResults)
+
+					// for _, scanResult := range scanResults {
+					// 	scanResults.ScaResults = append(scanResults.ScaResults, scanResult)
+					// }
+
+					workingDirs := []string{scanCmd.spec.Files[0].Pattern}
+					err = runner.RunJasScannersAndSetResults(&extendedScanResults, []coreutils.Technology{coreutils.Technology(scanResults.ScannedPackageType)}, []services.ScanResponse{*scanResults}, depsList, scanCmd.serverDetails, workingDirs, nil, false, "", applicability.ApplicabilityDockerScanScanType, secrets.SecretsScannerDockerScanType)
+
+					if err != nil {
+						log.Error(fmt.Sprintf("scanning '%s' failed with error: %s", graph.Id, err.Error()))
+						indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
+						return
+					}
+				}
+				resultsArr[threadId] = append(resultsArr[threadId], &ScanInfo{Target: filePath, Result: scanResults, ExtendedScanResults: &extendedScanResults})
```

Comment:
```suggestion
				extendedScanResults := utils.ExtendedScanResults{}
				if entitledForJas && scanCmd.commandSupportsJAS {
                    // Run Jas scans
					workingDirs := []string{filePath}
					err = runner.RunJasScannersAndSetResults(&extendedScanResults, []coreutils.Technology{coreutils.Technology(scanResults.ScannedPackageType)}, []services.ScanResponse{*scanResults}, depsListFromVulnerabilities(*scanResults), scanCmd.serverDetails, workingDirs, nil, false, "", applicability.ApplicabilityDockerScanScanType, secrets.SecretsScannerDockerScanType)

					if err != nil {
						log.Error(fmt.Sprintf("scanning '%s' failed with error: %s", graph.Id, err.Error()))
						indexedFileErrors[threadId] = append(indexedFileErrors[threadId], formats.SimpleJsonError{FilePath: filePath, ErrorMessage: err.Error()})
						return
					}
				}
				resultsArr[threadId] = append(resultsArr[threadId], &ScanInfo{Target: filePath, Result: scanResults, ExtendedScanResults: &extendedScanResults})
```
remove comments, pass the filepath and not pattern